lib/portage/package/ebuild/prepare_build_dirs.py

```diff
@@ -71,6 +72,7 @@ def makedirs(dir_path):
 
 	mydirs = [os.path.dirname(mysettings["PORTAGE_BUILDDIR"])]
 	mydirs.append(os.path.dirname(mydirs[-1]))
+	mydirs.append(empty_dir)
```

Comment:
We also need to delete it in the `bin/phase-functions.sh` `__dyn_clean`function or else it will be preserved (this requirement stems from a desire to detect stray files or directories created by ebuilds).